units/ftdc_rollups.go

```diff
@@ -19,111 +19,97 @@ const (
 	ftdcRollupsJobName = "ftdc-rollups"
 )
 
-type ftdcRollups struct {
+type ftdcRollupsJob struct {
 	PerfID       string              `bson:"perf_id" json:"perf_id" yaml:"perf_id"`
 	ArtifactInfo *model.ArtifactInfo `bson:"artifact" json:"artifact" yaml:"artifact"`
 
-	*job.Base `bson:"metadata" json:"metadata" yaml:"metadata"`
+	job.Base `bson:"metadata" json:"metadata" yaml:"metadata"`
 }
 
 func init() {
-	registry.AddJobType(ftdcRollupsJobName, func() amboy.Job {
-		fr := &ftdcRollups{}
-		fr.setup()
-		fr.SetDependency(dependency.NewAlways())
-
-		return fr
-	})
+	registry.AddJobType(ftdcRollupsJobName, func() amboy.Job { return makeFTDCRollupsJob() })
 }
 
-func (fr *ftdcRollups) setup() {
-	if fr.Base == nil {
-		fr.Base = &job.Base{
+func makeFTDCRollupsJob() *ftdcRollupsJob {
+	j := &ftdcRollupsJob{
+		Base: job.Base{
 			JobType: amboy.JobType{
 				Name:    ftdcRollupsJobName,
 				Version: 1,
 			},
-		}
+		},
 	}
-}
 
-func (fr *ftdcRollups) Validate() error {
-	fr.setup()
-	fr.SetDependency(dependency.NewAlways())
+	// TODO: Do I need to set priority ?
+	j.SetDependency(dependency.NewAlways())
+	return j
+}
 
-	if fr.PerfID == "" {
+func (j *ftdcRollupsJob) validate() error {
+	if j.PerfID == "" {
 		return errors.New("no id given")
 	}
 
-	if fr.ArtifactInfo == nil {
+	if j.ArtifactInfo == nil {
 		return errors.New("no artifact info given")
 	}
 
 	return nil
 }
 
-func (fr *ftdcRollups) reset() {
-	fr.ArtifactInfo = nil
-}
-
 func NewFTDCRollupsJob(perfId string, artifactInfo *model.ArtifactInfo) (amboy.Job, error) {
-	rollupJob := &ftdcRollups{
-		PerfID:       perfId,
-		ArtifactInfo: artifactInfo,
-	}
+	j := makeFTDCRollupsJob()
+	j.PerfID = perfId
+	j.ArtifactInfo = artifactInfo
 
-	if err := rollupJob.Validate(); err != nil {
-		return nil, errors.Wrap(err, "failed to setup new ftdc rollups job")
+	if err := j.validate(); err != nil {
+		return nil, errors.Wrap(err, "failed to create new ftdc rollups job")
 	}
-	return rollupJob, nil
+	return j, nil
 }
 
-func (fr *ftdcRollups) Run(ctx context.Context) {
-	defer fr.MarkComplete()
-	defer fr.reset()
+func (j *ftdcRollupsJob) Run(ctx context.Context) {
+	defer j.MarkComplete()
 
 	env := cedar.GetEnvironment()
```

Comment:
I sometimes stash the env in the job body and call GetEnvironment only when that's nil to theoretically make it mockable.